### ISSUE TITLE
SelfAttentionEstimator does not work with use_feat_dynamic_real = False

### ISSUE THREAD

**Issue by AmSpidey** [awslabs/gluonts#1976]
## Description
SelfAttentionEstimator does not seem to work with use_feat_dynamic_real = False.

I tried to use it to reproduce similar results from ones from the [Enhancing the Locality...](https://arxiv.org/abs/1907.00235) paper, but it seems that none of the datasets (electricity and traffic) have the feat_dynamic_real property. Plus the training does not run when I try to go around this issue by setting use_feat_dynamic_real to False.

## To Reproduce

```python
from gluonts.model import san
from gluonts.mx.trainer import Trainer
import gluonts.dataset.repository.datasets

dataset = gluonts.dataset.repository.datasets.get_dataset(
      dataset_name='traffic_nips')

trainer = Trainer(epochs=10)
estimator = san.SelfAttentionEstimator(
    freq=dataset.metadata.freq, prediction_length=dataset.metadata.prediction_length, 
    context_length=100,
    use_feat_dynamic_real = False,
    trainer=trainer)
predictor = estimator.train(training_data=dataset.train)
```

## Error message or code output

```
/usr/local/lib/python3.7/dist-packages/gluonts/json.py:102: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  "Using `json`-module for json-handling. "
/usr/local/lib/python3.7/dist-packages/gluonts/dataset/common.py:324: FutureWarning: The 'freq' argument in Timestamp is deprecated and will be removed in a future version.
  timestamp = pd.Timestamp(timestamp_input, freq=freq)
/usr/local/lib/python3.7/dist-packages/gluonts/dataset/common.py:327: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  if isinstance(timestamp.freq, Tick):
/usr/local/lib/python3.7/dist-packages/gluonts/dataset/common.py:329: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  timestamp.floor(timestamp.freq), timestamp.freq
/usr/local/lib/python3.7/dist-packages/gluonts/dataset/common.py:329: FutureWarning: The 'freq' argument in Timestamp is deprecated and will be removed in a future version.
  timestamp.floor(timestamp.freq), timestamp.freq
saving time-series into /root/.mxnet/gluon-ts/datasets/traffic_nips/train/data.json
saving time-series into /root/.mxnet/gluon-ts/datasets/traffic_nips/test/data.json
/usr/local/lib/python3.7/dist-packages/gluonts/transform/feature.py:343: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  self._freq_base = start.freq.base
/usr/local/lib/python3.7/dist-packages/gluonts/transform/split.py:36: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  return _shift_timestamp_helper(ts, ts.freq, offset)
/usr/local/lib/python3.7/dist-packages/gluonts/transform/feature.py:386: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  if self._full_range_date_features is not None
/usr/local/lib/python3.7/dist-packages/gluonts/transform/feature.py:340: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version
  self._freq_base is None or self._freq_base == start.freq.base
  0%|          | 0/50 [00:00<?, ?it/s]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-2-42fb1bd94255>](https://localhost:8080/#) in <module>()
     12     use_feat_dynamic_real = False,
     13     trainer=trainer)
---> 14 predictor = estimator.train(training_data=dataset.train)

11 frames
[/usr/local/lib/python3.7/dist-packages/gluonts/mx/model/estimator.py](https://localhost:8080/#) in train(self, training_data, validation_data, num_workers, num_prefetch, shuffle_buffer_length, cache_data, **kwargs)
    198             num_prefetch=num_prefetch,
    199             shuffle_buffer_length=shuffle_buffer_length,
--> 200             cache_data=cache_data,
    201         ).predictor

[/usr/local/lib/python3.7/dist-packages/gluonts/mx/model/estimator.py](https://localhost:8080/#) in train_model(self, training_data, validation_data, num_workers, num_prefetch, shuffle_buffer_length, cache_data)
    170             net=training_network,
    171             train_iter=training_data_loader,
--> 172             validation_iter=validation_data_loader,
    173         )
    174 

[/usr/local/lib/python3.7/dist-packages/gluonts/mx/trainer/_base.py](https://localhost:8080/#) in __call__(self, net, train_iter, validation_iter)
    437                     epoch_no,
    438                     train_iter,
--> 439                     num_batches_to_use=self.num_batches_per_epoch,
    440                 )
    441 

[/usr/local/lib/python3.7/dist-packages/gluonts/mx/trainer/_base.py](https://localhost:8080/#) in loop(epoch_no, batch_iter, num_batches_to_use, is_training)
    332                     if first_forward:
    333                         first_forward = False
--> 334                         _ = net(*batch.values())
    335 
    336                         self.callbacks.on_network_initializing_end(

[/usr/local/lib/python3.7/dist-packages/mxnet/gluon/block.py](https://localhost:8080/#) in __call__(self, *args)
    823             hook(self, args)
    824 
--> 825         out = self.forward(*args)
    826 
    827         for hook in self._forward_hooks.values():

[/usr/local/lib/python3.7/dist-packages/mxnet/gluon/block.py](https://localhost:8080/#) in forward(self, x, *args)
   1490                                      'Find all contexts = {}'.format(ctx_set))
   1491                 with ctx:
-> 1492                     return self._call_cached_op(x, *args)
   1493             with ctx:
   1494                 try:

[/usr/local/lib/python3.7/dist-packages/mxnet/gluon/block.py](https://localhost:8080/#) in _call_cached_op(self, *args)
   1198     def _call_cached_op(self, *args):
   1199         if self._cached_op is None:
-> 1200             self._build_cache(*args)
   1201         assert self._cached_op, "Gluon failed to build the cache. " \
   1202                                 "This should never happen. " \

[/usr/local/lib/python3.7/dist-packages/mxnet/gluon/block.py](https://localhost:8080/#) in _build_cache(self, *args)
   1066 
   1067     def _build_cache(self, *args):
-> 1068         data, out = self._get_graph(*args)
   1069         data_names = {data.name: i for i, data in enumerate(data)}
   1070         input_names = out.list_inputs()

[/usr/local/lib/python3.7/dist-packages/mxnet/gluon/block.py](https://localhost:8080/#) in _get_graph(self, *args)
   1058             params = {i: j.var() for i, j in self._reg_params.items()}
   1059             with self.name_scope():
-> 1060                 out = self.hybrid_forward(symbol, *grouped_inputs, **params)  # pylint: disable=no-value-for-parameter
   1061             out, self._out_format = _flatten(out, "output")
   1062 

[/usr/local/lib/python3.7/dist-packages/gluonts/model/san/_network.py](https://localhost:8080/#) in hybrid_forward(self, F, past_target, past_observed_values, past_is_pad, future_target, future_observed_values, past_feat_dynamic_real, past_feat_dynamic_cat, future_feat_dynamic_real, future_feat_dynamic_cat, feat_static_real, feat_static_cat)
    344             future_feat_dynamic_cat,
    345             feat_static_real,
--> 346             feat_static_cat,
    347         )
    348 

[/usr/local/lib/python3.7/dist-packages/gluonts/model/san/_network.py](https://localhost:8080/#) in _preprocess(self, F, past_target, past_observed_values, past_is_pad, past_feat_dynamic_real, past_feat_dynamic_cat, future_target, future_feat_dynamic_real, future_feat_dynamic_cat, feat_static_real, feat_static_cat)
    257             feat_static_real,
    258             feat_static_cat,
--> 259             is_past=True,
    260         )
    261         future_covariates = _assemble_covariates(

[/usr/local/lib/python3.7/dist-packages/gluonts/model/san/_network.py](https://localhost:8080/#) in _assemble_covariates(feat_dynamic_real, feat_dynamic_cat, feat_static_real, feat_static_cat, is_past)
    208         ) -> Tensor:
    209             covariates = []
--> 210             if feat_dynamic_real.shape[-1] > 0:
    211                 covariates.append(feat_dynamic_real)
    212             if feat_static_real.shape[-1] > 0:

AttributeError: 'Symbol' object has no attribute 'shape'
```

## Environment
Colab
- Python version: 3.7.13
- GluonTS version: gluonts-0.9.4
- MXNet version: mxnet-1.9.0

**Comment by lostella** [comment]
When turning off hybridization with
```python
trainer = Trainer(epochs=10, hybridize=False)
```

I'm further getting

```
  File "/Users/stellalo/gluon-ts/src/gluonts/model/san/_network.py", line 241, in _assemble_covariates
    embeddings = self.embedder(categories)
AttributeError: 'SelfAttentionTrainingNetwork' object has no attribute 'embedder'
```

**Comment by jaheba** [comment]
@lostella was this fixed by #1994?

**Comment by lostella** [comment]
Model being moved to `gluonts.nursery` in #2534
